File: templates/Uwp/Features/ShareTarget._VB/ViewModels/SharedDataViewModelBase.vb
Imports Windows.ApplicationModel.DataTransfer.ShareTarget

Namespace ViewModels
    Public Class SharedDataViewModelBase
        Inherits System.ComponentModel.INotifyPropertyChanged

        Private _dataFormat As String

        Public Property DataFormat As String
            Get
                Return _dataFormat
            End Get
            Set
                [Param_Setter](_dataFormat, value)
            End Set
        End Property

        Private _pageTitle As String

        Public Property PageTitle As String
            Get
                Return _pageTitle
            End Get
            Set
                [Param_Setter](_pageTitle, value)
            End Set
        End Property

        Private _title As String

        Public Property Title As String
            Get
                Return _title
            End Get
            Set
                [Param_Setter](_title, value)
            End Set
        End Property

        Public Sub New()
        End Sub

        Public Overridable Async Function LoadDataAsync(shareOperation As ShareOperation) As Task
            Title = shareOperation.Data.Properties.Title
            Await Task.CompletedTask
        End Function
    End Class
End Namespace

File: code/TemplateStudioForUWP/Templates/Uwp/Features/ShareTarget/ViewModels/SharedDataViewModelBase.cs
using System;
using System.Threading.Tasks;
using Windows.ApplicationModel.DataTransfer.ShareTarget;

namespace Param_RootNamespace.ViewModels
{
    public class SharedDataViewModelBase : System.ComponentModel.INotifyPropertyChanged
    {
        private string _dataFormat;

        public string DataFormat
        {
            get => _dataFormat;
            set => Param_Setter(ref _dataFormat, value);
        }

        private string _pageTitle;

        public string PageTitle
        {
            get => _pageTitle;
            set => Param_Setter(ref _pageTitle, value);
        }

        private string _title;

        public string Title
        {
            get => _title;
            set => Param_Setter(ref _title, value);
        }

        public SharedDataViewModelBase()
        {
        }

        public virtual async Task LoadDataAsync(ShareOperation shareOperation)
        {
            Title = shareOperation.Data.Properties.Title;
            await Task.CompletedTask;
        }
    }
}
